research/context/analysis/art.py

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Conversation-AI.github.io Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+approximate randomization test (art)
+
+It is based on the approximate randomization test used to compare the superiority of a system against
+a baseline (see https://cs.stanford.edu/people/wmorgan/sigtest.pdf).
+
+This code measures the statistical significance of the claim that the agreement in one annotation job is higher
+than the other. The latter is the baseline and it is expected to be harder for the annotators (thus, lower agreement).
+Following, in comments, the former is called easier-task while the latter harder-task (baseline).
+
+This test assumes that the two annotation jobs:
+a) contain judgments from two distinct, non-overlapping groups of annotators (as in AB testing), and
+b) the same comments are given to the annotators of the two groups and judgments are 1:1 between the two groups
+
+The returned values are:
+1) the difference in agreement between the two annotation tasks (the higher the better), and
+2) the p-value (the lower the better).
+
+Run as:
+
+python art.py \
+--et_coders "et_coders.json" \
+--et_judgments "et_labels.json" \
+--ht_coders "ht_coders.json" \
+--ht_judgments "ht_labels.json" \
+--repetitions 1000 \
+
+Note that one can also use "gs://path.json" file paths.
+"""
+
+import tensorflow as tf
+import numpy as np
+import json
+import krippendorff
+from sklearn.metrics import accuracy_score
+
+FLAGS = tf.app.flags.FLAGS
+tf.app.flags.DEFINE_integer("repetitions", 1000, "Number of samples to be performed.")
+tf.app.flags.DEFINE_string("et_coders", None, "JSON file with easier-task coders UIDs - required for Krippendorff's alpha")
```

Comment:
can you do pyformat -i maybe, would be nice to wrap long lines.
(Git doesn't even have a scroll bar and doesn't resize wider than ~100 cols.)

Reply:
Nice tool, sure! Just pushed a commit with code formatted.